src/main/java/fr/gravendev/multibot/moderation/events/GuildMessageReceivedListener.java

```diff
@@ -18,16 +18,23 @@
 import net.dv8tion.jda.api.entities.TextChannel;
 import net.dv8tion.jda.api.entities.User;
 import net.dv8tion.jda.api.events.message.guild.GuildMessageReceivedEvent;
+import org.eclipse.jetty.util.thread.ScheduledExecutorScheduler;
 
 import java.awt.*;
 import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
 
 public class GuildMessageReceivedListener implements Listener<GuildMessageReceivedEvent> {
 
     private final BadWordsDAO badWordsDAO;
     private final InfractionDAO infractionDAO;
     private final ImmunisedIdDAO immunisedIdsDAO;
 
+    private final ScheduledExecutorScheduler scheduledExecutorScheduler = new ScheduledExecutorScheduler();
+    private final Map<String, Integer> spamCounter = new HashMap<>();
```

Comment:
HashMap is not thread-safe, maybe you can encapsulate it within `Collections.synchronizedMap(...)`.